src/Microsoft.OData.Core/UriParser/Binders/InBinder.cs

```diff
@@ -92,6 +94,39 @@ private CollectionNode GetCollectionOperandFromToken(QueryToken queryToken, IEdm
                     replacedText[0] = '[';
                     replacedText[replacedText.Length - 1] = ']';
                     bracketLiteralText = replacedText.ToString();
+
+                    Debug.Assert(expectedType.IsCollection());
+                    if (expectedType.Definition.AsElementType().FullTypeName().Equals("Edm.String"))
+                    {
+                        // For collection of strings, need to convert single-quoted string to double-quoted string,
+                        // and also, per ABNF, two consecutive single quotes  to one single quote.
+                        // Sample: ['a''bc','''def','xyz'''] ==> ["a'bc","'def","xyz'"], which is legitimate Json format.
+                        string[] items = bracketLiteralText.Substring(1, bracketLiteralText.Length - 2).Split(',');
+
+                        // Skip conversion if the items are already in double-quote format (for backward compatibility).
+                        // Note that per ABNF, query option strings should use single quotes.
+                        if (items.Length > 0 && items[0][0] == '\'')
+                        {
+                            StringBuilder builder = new StringBuilder();
+                            for (int i = 0; i < items.Length; i++)
+                            {
+                                string convertedItem = UriParserHelper.RemoveQuotes(items[i]);

```

Comment:
What happens if the client is trying to find strings that actually have single quotes? For example, `StringItem in ('Alan's computer', 'Biao's mouse', 'Sam's keyboard')`.

Reply:
Then the single-quote should be escaped by using two consecutive single-quotes, per ABNF.
> string           = SQUOTE *( SQUOTE-in-string / pchar-no-SQUOTE ) SQUOTE
SQUOTE-in-string = SQUOTE SQUOTE ; two consecutive single quotes represent one within a string literal

For example: ('Alan''s computer', 'Biao''s mouse', 'Sam''s keyboard')

Reply:
ah I see, so this was indeed a bug. Thanks for the pointing this out from the ABNF!